### ISSUE TITLE
Unhandled Exception: type 'int' is not a subtype of type 'String'

### ISSUE THREAD

**Issue by rokk4** [livekit/client-sdk-flutter#116]
In the example app, this shows up in the logs.

```
[VERBOSE-2:ui_dart_state.cc(209)] Unhandled Exception: type 'int' is not a subtype of type 'String'
#0      RTCDataChannelNative.eventListener (package:flutter_webrtc/src/native/rtc_data_channel_impl.dart:54:42)
#1      _rootRunUnary (dart:async/zone.dart:1434:47)
#2      _CustomZone.runUnary (dart:async/zone.dart:1335:19)
#3      _CustomZone.runUnaryGuarded (dart:async/zone.dart:1244:7)
#4      _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:341:11)
#5      _DelayedData.perform (dart:async/stream_impl.dart:591:14)
#6      _StreamImplEvents.handleNext (dart:async/stream_impl.dart:706:11)
#7      _PendingEvents.schedule.<anonymous closure> (dart:async/stream_impl.dart:663:7)
#8      _rootRun (dart:async/zone.dart:1418:47)
#9      _CustomZone.run (dart:async/zone.dart:1328:19)
#10     _CustomZone.runGuarded (dart:async/zone.dart:1236:7)
#11     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1276:23)

#12     _rootRun (dart:async/zone.dart:1426:13)
#13     _CustomZone.run (dart:async/zone.dart:1328:19)
#14     _CustomZone.runGuarded (dart:async/zone.dart:1236:7)
#15     _CustomZone.bindCallbackGuarded.<anonymous closure> (dart:async/zone.dart:1276:23)
#16     _microtaskLoop (dart:async/schedule_microtask.dart:40:21)
#17     _startMicrotaskLoop (dart:async/schedule_microtask.dart:49:5)
```

Where does it come from?
Can it be ignored?

Steps to reproduce:
use example app on iphone with 4G/5G
server URLwss://live. txxhexxraxxpy-lxxixxft.net (without the xxs because SEO ...)
token: eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE2NTQ3ODY4NTAsImlzcyI6IkFQSWFlb0JFeENOdG5CUyIsIm5iZiI6MTY1MjE5NDg1MCwic3ViIjoic2wiLCJ2aWRlbyI6eyJyb29tIjoidGVzdDAxIiwicm9vbUpvaW4iOnRydWV9fQ.7m6xVj3UFXIZfAucgqLjziZEefuX8ZLmgAMPYnEAWVE

I have attached flutter and server logs.

[flutter_SUCCESS.txt](https://github.com/livekit/client-sdk-flutter/files/8673247/flutter_SUCCESS.txt)
[server_SUCCESS.txt](https://github.com/livekit/client-sdk-flutter/files/8673248/server_SUCCESS.txt)

**Comment by hiroshihorie** [comment]
Hi, I think this was fixed in the dependency we use.
Can you try `flutter pub upgrade` and compiling it again.

**Comment by rokk4** [comment]
@hiroshihorie thanks for the reply :)

On Android the issue seems fixed by using the latest commit directly from git.
Also I am using Flutter 3.0 now.

However now two new issues came up, so I could only check it on Android.
On MacOS the Camera is no longer being recognized.

And iOS now crashes like this directly on connection:
```
flutter:
flutter: video track TR_VCcxBxNypMfdNU subscribed false muted false
[VERBOSE-3:platform_message_handler_ios.mm(106)] Check failed: task_runners_.GetPlatformTaskRunner()->RunsTasksOnCurrentThread().
* thread #26, name = 'signaling_thread 0x0x104621320', stop reason = signal SIGABRT
    frame #0: 0x00000001cde28b78 libsystem_kernel.dylib`__pthread_kill + 8
libsystem_kernel.dylib`__pthread_kill:
->  0x1cde28b78 <+8>:  b.lo   0x1cde28b98               ; <+40>
    0x1cde28b7c <+12>: pacibsp 
    0x1cde28b80 <+16>: stp    x29, x30, [sp, #-0x10]!
    0x1cde28b84 <+20>: mov    x29, sp
Target 0: (Runner) stopped.
Lost connection to device.
```

**Comment by hiroshihorie** [comment]
Yes I think this is fixed in the PR I submitted to `flutter_webrtc` package.

**Comment by hiroshihorie** [comment]
This should be fixed in v1.0
